### ISSUE TITLE
NlSocketHandle::iter() shouldn't force use of NlTypeWrapper

### ISSUE THREAD

**Issue by joshtriplett** [jbaublitz/neli#108]
`NlTypeHandle` is a convenient abstraction for when you need to parse a message and don't know what kind of `nl_type` value to expect.

However, if you have a specific type of message to parse, it would be more convenient to directly get an `Nlmsghdr<T, P>` where `T` is one of the specific types like `Rtm` that implements `NlType`.

Unfortunately, `NlSocketHandle::iter()` only can only provide an iterator over `Nlmsghdr<NlTypeWrapper, P>`.

`NlMessageIter` partly supports iteration even without using `NlTypeWrapper`, but only implements `Iterator` if using `NlTypeWrapper`. It should be possible for `NlMessageIter` to always implement `Iterator` as long as it has an `NlType`.

**Comment by jbaublitz** [comment]
When I made this decision, this was prior to @phip1611's feedback on using neli with custom netlink families that will not be contributed upstream. I made the faulty assumption that most people would just want to contribute their families upstream, but this was operating under the assumption that these families would be generally applicable and included in the Linux kernel. Given that I already do have users using this for custom netlink families that are not generally applicable and will not be submitted for upstream inclusion, I agree that I should reconsider this. Given that I don't want to break backwards compatibility, I'm going to add a new method that accepts both type parameters and add a deprecation notice to the old iter method that in will be removed in the next minor release. How does that sound?
